src/services/crawl-tx/crawl_tx.service.ts

```diff
@@ -586,15 +586,19 @@ export default class CrawlTxService extends BullableService {
     attributeKey: string
   ): string {
     let result = '';
-    events.forEach((event: any) => {
-      if (result === '' && event.type === eventType) {
-        event.attributes.forEach((attribute: any) => {
-          if (result === '' && attribute.key === attributeKey) {
-            result = attribute.value;
-          }
-        });
-      }
-    });
+    const foundEvent = events.find(
```

Comment:
hmm, nhẽ ra thì for thường xong rồi return luôn cũng được đúng ko? cơ mà để sau cũng được, tách hẳn cái hàm này cho vào ultility hay helper gì đó